File: hive_client_extensions/cpp/hiveLogic.cpp
#include "hiveLogic.hpp"

namespace AI {
    HiveLogic::HiveLogic() {}
    HiveLogic::~HiveLogic() {}
    Hive::Move HiveLogic::GetNextMove(Hive::GameState currentGameState, Hive::Color ownPlayerColor) {
        srand (time(NULL));
        std::vector<Hive::Move> possibleMoves = currentGameState.GetPossibleMoves();
        return possibleMoves[rand() % possibleMoves.size()];
    }

    void HiveLogic::OnGameEnd(Hive::Color colorOfWinningPlayer) {
        std::cout << "Winning player has the color: " + Hive::ColorToString(colorOfWinningPlayer) + "\n";
    }
}  // namespace AI

File: hive_client_extensions/python/HiveLogic.py
import Hive, random

class HiveLogic(Hive.Logic):
    def __init__(self):
        Hive.Logic.__init__(self)
    
    def GetNextMove(self, currentGameState: "Hive::GameState", ownPlayerColor: "Hive::Color")  -> "Hive::Move":
        print("Hello there.")
        return random.choice(currentGameState.GetPossibleMoves())

    def OnGameEnd(self, colorOfWinningPlayer: "Hive::Color"):
        print("Winning player has the color: " + Hive.ColorToString(colorOfWinningPlayer))